clusters/clusters_api.go

```diff
@@ -611,6 +612,14 @@ func wrapMissingClusterError(err error, id string) error {
 	if apiErr.IsMissing() {
 		return err
 	}
+	// https://github.com/databrickslabs/terraform-provider-databricks/issues/1177
+	// Aligned with Clusters Core team to keep behavior of these workarounds 
+	// as is in the longer term, so that this keeps working.
+	if apiErr.ErrorCode == "INVALID_STATE" {
```

Comment:
Add the expectation that the return code is 400?

Reply:
yep, that's our `INVALID_STATE` mapping. i'd rather have it as `NOT_FOUND` (imho, this is more semantically correct, as the cluster is **_no longer found_**), so that we don't have to have this if. you're on that slack thread.

Reply:
So any response with status 400 gets `ErrorCode = "INVALID_STATE"`?